### PR TITLE
Updated cli instructions with additonal details for each subcommands

### PR DESCRIPTION
**Pull request by varunfb**

### CODE REVIEW

File: docs/cli_reference.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 The `llama` CLI tool helps you setup and use the Llama toolchain & agentic systems. It should be available on your path after installing the `llama-toolchain` package.
 
+### Subcommands 
+1. `download`: `llama` cli tools supports downloading the model from Meta or HuggingFace. 
+2. `model`: Shows additional properties of the llama models.
```
Comment (ashwinb) [inline docs/cli_reference.md:7]: can we say "Lists available models and their properties."

File: docs/cli_reference.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+3. `distribution`: A distribution is a set of REST API, this command allows you to manage (list, install, create, configure, start) distributions. You can read more about this here.
```
Comment (ashwinb) [inline docs/cli_reference.md:8]: set of REST APIs (plural)

Comment (ashwinb) [reply docs/cli_reference.md:8]: where is "about this here" pointing to?

**Review by ashwinb — APPROVED** [review]

File: docs/cli_reference.md (hunk 2 above)
Comment (varunfb) [reply docs/cli_reference.md:8]: I need the hyperlink to be created for the sub-headers first and then add the link in the next PR. Will take care of the fixes in the next PR.

**Pull request merged**